modules/siddhi-core/src/main/java/org/wso2/siddhi/core/aggregation/IncrementalDataPurging.java

```diff
@@ -0,0 +1,307 @@
+
+    public void init(AggregationDefinition aggregationDefinition, StreamEventPool streamEventPool,
+                     Map<TimePeriod.Duration, Table> aggregationTables, Boolean isProcessingOnExternalTime,
+                     SiddhiAppContext siddhiAppContext) {
+        this.siddhiAppContext = siddhiAppContext;
+        List<Annotation> annotations = aggregationDefinition.getAnnotations();
+        this.streamEventPool = streamEventPool;
+        this.aggregationTables = aggregationTables;
+        if (isProcessingOnExternalTime) {
+            purgingTimestampField = EXTERNAL_AGG_TIMESTAMP_FIELD;
+        } else {
+            purgingTimestampField = INTERNAL_AGG_TIMESTAMP_FIELD;
+        }
+        aggregatedTimestampAttribute = new Attribute(purgingTimestampField, Attribute.Type.LONG);
+
+        VariableExpressionExecutor variableExpressionExecutor = new VariableExpressionExecutor(
+                aggregatedTimestampAttribute, 0, 1);
+        variableExpressionExecutorList.add(variableExpressionExecutor);
+        for (Map.Entry<TimePeriod.Duration, Table> entry : aggregationTables.entrySet()) {
+            this.tableMap.put(entry.getValue().getTableDefinition().getId(), entry.getValue());
+            switch (entry.getKey()) {
+                case SECONDS:
+                    retentionPeriods.put(entry.getKey(), Expression.Time.sec(120).value());
+                    minimumDurationMap.put(entry.getKey(), Expression.Time.sec(120).value());
+                    break;
+                case MINUTES:
+                    retentionPeriods.put(entry.getKey(), Expression.Time.hour(24).value());
+                    minimumDurationMap.put(entry.getKey(), Expression.Time.minute(120).value());
+                    break;
+                case HOURS:
+                    retentionPeriods.put(entry.getKey(), Expression.Time.day(30).value());
+                    minimumDurationMap.put(entry.getKey(), Expression.Time.hour(25).value());
+                    break;
+                case DAYS:
+                    retentionPeriods.put(entry.getKey(), Expression.Time.year(1).value());
+                    minimumDurationMap.put(entry.getKey(), Expression.Time.day(32).value());
+                    break;
+                case MONTHS:
+                    retentionPeriods.put(entry.getKey(), RETAIN_ALL);
+                    minimumDurationMap.put(entry.getKey(), Expression.Time.month(13).value());
+                    break;
+                case YEARS:
+                    retentionPeriods.put(entry.getKey(), RETAIN_ALL);
+                    minimumDurationMap.put(entry.getKey(), 0L);
+            }
+        }
+
+        Map<String, Annotation> annotationTypes = new HashMap<>();
+
+        for (Annotation annotation : annotations) {
+            annotationTypes.put(annotation.getName().toLowerCase(), annotation);
+        }
+        Annotation purge = annotationTypes.get(SiddhiConstants.NAMESPACE_PURGE);
+        if (Objects.nonNull(purge)) {
```

Comment:
Simply use != instead. `Objects.nonNull` is there to be used as a predicate.

Reply:
Fixed by 6759878